examples/PlaneCurve.jl

```diff
@@ -0,0 +1,273 @@
+module PlaneCurveModule
+using Oscar, Markdown
+import Base.:(==)
+
+export Point, ideal_point, AffinePlaneCurve, ProjPlaneCurve, hash, degree,
+       jacobi_ideal, curve_components, isirreducible, isreduced, reduction,
+       union
+
+################################################################################
+
+abstract type PlaneCurve end
+
+################################################################################
+# Point (not specific to curves).
+################################################################################
+# Point (first attempt).
+# The point is described by its coordinates.
+# ambient_dim gives the dimension of the space to which belongs the point.
+
+mutable struct Point{S}

```

Comment:
Is `S` again meant to be the type of a field element? Then a type annotation saying so would be helpful:
```suggestion
mutable struct Point{S <: FieldElem}
```